File: phonebridge-common/src/main/java/com/phonebridge/row/campaign/Campaign.java
package com.phonebridge.row.campaign;

import com.phonebridge.row.BasePojo;
import com.phonebridge.row.account.Account;

import lombok.Data;

@Data
public class Campaign extends BasePojo{

	private Account account;
	private String campaignId;
	private String campaignName;
	private EnumAssocDirection callDirection;
	private EnumStatus campaignStatus;
	private boolean isEnabledForPopup;
	/* Defines the type of Campaign */
	private EnumTypeOfCall typeOfCall;
	/* Default campaigns are generic those which don't fit into other campaigns */
	private boolean isDefault;
	
	private CampaignIncomingSettings campaignIncomingSettings;
	private CampaignOutgoingSettings campaignOutgoingSettings;
	private CampaignPopup campaignPopup;
}

File: phone-bridge-ui/src/app/layout/campaign/model/Campaign.ts
import { CampaignIncomingSettings } from './CampaignIncomingSettings';
import { CampaignOutgoingSettings } from './CampaignOutgoingSettings';
import { CampaignPopup } from './CampaignPopup';

export class Campaign {
    campaignName: string;
    callDirection: string;
    campaignStatus: string;
    isEnabledForPopup: boolean;
    /* Defines the type of Campaign */
    typeOfCall: string;
    /* Default campaigns are generic those which don't fit into other campaigns */
    isDefault: boolean;

    campaignIncomingSettings: CampaignIncomingSettings;
    campaignOutgoingSettings: CampaignOutgoingSettings;
    campaignPopup: CampaignPopup;
}

